js/whitebit.js

```diff
@@ -615,11 +707,440 @@ module.exports = class whitebit extends Exchange {
+            memo = this.safeValue (addressField, 'memo');
+        }
+        this.checkAddress (address);
+        return {
+            'currency': code,
+            'address': address,
+            'tag': memo,
+            'network': this.safeValue (params, 'network'),
+            'info': response,
+        };
+    }
+
+    async withdraw (code, amount, address, tag = undefined, params = {}) {
+        await this.loadMarkets ();
+        const currency = this.currency (code); // check if it has canDeposit
+        const request = {
+            'ticker': currency['id'],
+            'amount': this.numberToString (amount),
+            'address': address,
+        };
+        const uniqueId = this.safeValue (params, 'uniqueId');
+        if (uniqueId === undefined) {
+            throw new ArgumentsRequired (this.id + ' fetchDepositAddress() requires an uniqueId');
+        }
+        request['uniqueId'] = uniqueId;
+        if (tag !== undefined) {
+            request['memo'] = tag;
+        }
+        if (this.isFiat (code)) {
+            const provider = this.safeValue (params, 'provider');
+            if (provider === undefined) {
+                throw new ArgumentsRequired (this.id + ' fetchDepositAddress() requires a provider when the ticker is fiat');
+            }
+            request['provider'] = provider;
+        }
+        const response = await this.v4PrivatePostMainAccountWithdraw (this.extend (request, params));
+        //
+        // [
+        //   empty array - has success status - go to deposit/withdraw history and check you request status by uniqueId
+        // ]
+        //
+        return {
+            'id': uniqueId,
+            'info': response,
+        };
+    }
+
+    isFiat (currency) {
+        const fiatCurrencies = this.safeValue (this.options, 'fiatCurrencies', []);
+        return this.inArray (currency, fiatCurrencies);
+    }
+
+    sign (path, api = 'public', method = 'GET', params = {}, headers = undefined, body = undefined) {
         const query = this.omit (params, this.extractParams (path));
-        let url = this.urls['api'][api] + '/' + this.implodeParams (path, params);
-        if (Object.keys (query).length) {
-            url += '?' + this.urlencode (query);
+        const version = this.safeValue (api, 0);
+        const accessibility = this.safeValue (api, 1);
+        let url = this.urls['api'][version][accessibility] + '/' + this.implodeParams (path, params);
+        if (accessibility === 'public') {
+            if (Object.keys (query).length) {
+                url += '?' + this.urlencode (query);
+            }
+        }
+        if (accessibility === 'private') {
+            this.checkRequiredCredentials ();
+            const nonce = this.nonce ().toString ();
+            const secret = this.stringToBinary (this.secret);
+            const baseUrl = this.urls['api']['web'];
+            const request = '/' + url.replace (baseUrl, '');
```

Comment:
In this case, we are removing the baseUrl from the already concatenated url, i would suggest to do this otherwise: we never concatenate strings just to cut them out later. Intead we should add conditionals that would only prepend `this.urls['api'][version][accessibility]` to the `url` after the `'private'` section.

Reply:
Done